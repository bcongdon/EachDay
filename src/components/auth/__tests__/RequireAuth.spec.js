import React, { Component } from 'react'
import RequireAuth from '../RequireAuth'
import { mount } from 'enzyme'
import { PropTypes } from 'prop-types'
import toJson from 'enzyme-to-json'
import configureStore from 'redux-mock-store'
const middlewares = []
const mockStore = configureStore(middlewares)

class TestComponent extends Component {
  render () {
    return <p>Testing</p>
  }
}

it('should push to login page without authentication', () => {
  const Composed = RequireAuth(TestComponent)
  const history = jest.fn()
  history.push = jest.fn()

  const context = {
    router: { history }
  }
  const childContextTypes = {
    router: PropTypes.object
  }

  var reqAuth = mount(
    <Composed store={mockStore({auth: {authenticated: false}})} />,
    { context, childContextTypes }
  )
  expect(history.push).toBeCalled()
  expect(toJson(reqAuth)).toMatchSnapshot()
})

it('should not change routes when user is authenticated', () => {
  const Composed = RequireAuth(TestComponent)
  const history = jest.fn()
  history.push = jest.fn()

  const context = {
    router: { history }
  }
  const childContextTypes = {
    router: PropTypes.object
  }

  var reqAuth = mount(
    <Composed store={mockStore({auth: {authenticated: true}})} />,
    { context, childContextTypes }
  )
  expect(history.push).not.toBeCalled()
  expect(toJson(reqAuth)).toMatchSnapshot()
})
