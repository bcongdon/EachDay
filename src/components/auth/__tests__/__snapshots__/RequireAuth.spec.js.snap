// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not change routes when user is authenticated 1`] = `
<Connect(Authentication)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Authentication
    authenticated={true}
    dispatch={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  >
    <TestComponent
      authenticated={true}
      dispatch={[Function]}
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
      storeSubscription={
        Subscription {
          "listeners": Object {
            "clear": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": [Function],
        }
      }
    >
      <p>
        Testing
      </p>
    </TestComponent>
  </Authentication>
</Connect(Authentication)>
`;

exports[`should push to login page without authentication 1`] = `
<Connect(Authentication)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Authentication
    authenticated={false}
    dispatch={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  >
    <TestComponent
      authenticated={false}
      dispatch={[Function]}
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
      storeSubscription={
        Subscription {
          "listeners": Object {
            "clear": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": [Function],
        }
      }
    >
      <p>
        Testing
      </p>
    </TestComponent>
  </Authentication>
</Connect(Authentication)>
`;
